Fix grid not getting invalidated when changing the Height/Width of Row/ColumnDefinitions declared with the short syntax

OvrBtn opened the pull request:
### Description of Change

### Issues Fixed

Fixes #25983

### After change

https://github.com/user-attachments/assets/5a1c2e65-ba4b-419f-928a-a405dc74d504

OvrBtn committed:
Subscribe to `SizeChanged` in constructor

jsuarezruiz reviewed (commented):
Could include an UITest?

OvrBtn committed:
Add UI test

mattleibow commented:
Might not need a uitest, could be a unit test. I wonder if some of the existing test check for the update and can be copied for this scenario.

kubaflo committed:
Improve UI tests with programmatic assertions

- Rename Issues25983 -> Issue25983 to follow naming convention
- Replace VerifyScreenshot() with programmatic assertions that measure element sizes
- Split into two separate tests: GridRowDefinitionResizeWithShortSyntax and GridColumnDefinitionResizeWithShortSyntax
- Tests verify element rect height/width increases after button clicks
